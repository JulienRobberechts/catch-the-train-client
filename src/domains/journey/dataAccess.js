import missions from "../../data/ratp/rers/A/missions.json";

export function importMissionsCodes() {
  return missions;
}

export function importMissionsSchedules(missionCodes) {
  const missionsSchedules = missionCodes.map((missionCode) => ({
    mission: missionCode,
    ...require(`../../data/ratp/rers/A/stations-${missionCode}.json`),
  }));
  return missionsSchedules;
}
