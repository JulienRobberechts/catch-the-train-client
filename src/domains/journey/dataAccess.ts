import missions from "../../data/ratp/rers/A/missions.json";
import stations from "../../data/ratp/rers/A/stations.json";
import { MissionStationDefinition } from "../../data/ratp/rers/A/types";

export function importMissionsCodes() {
  return missions;
}

export interface MissionsSchedule {
  mission: string;
  result: MissionStationDefinition;
}

export function importMissionsSchedules(
  missionCodes: string[]
): MissionsSchedule[] {
  const missionsSchedules = missionCodes.map((missionCode) => ({
    mission: missionCode,
    ...require(`../../data/ratp/rers/A/stations-${missionCode}.json`),
  }));
  return missionsSchedules;
}

export function importStations() {
  return stations;
}
