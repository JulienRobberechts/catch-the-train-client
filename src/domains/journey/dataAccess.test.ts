import { importMissionsCodes, importMissionsSchedules } from "./dataAccess";

describe("dataAccess", () => {
  describe("importMissionsCodes", () => {
    test("should import Missions codes", () => {
      const missionCodes = importMissionsCodes();
      expect(missionCodes).toMatchSnapshot();
    });
  });
  describe("importMissionsSchedules", () => {
    test("should import Missions Schedules", () => {
      const missionCodes = ["NANI", "NOTE"];
      const missionsSchedules = importMissionsSchedules(missionCodes);
      expect(missionsSchedules).toMatchSnapshot();
    });
  });
});
