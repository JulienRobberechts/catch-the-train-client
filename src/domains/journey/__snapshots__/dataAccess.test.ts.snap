// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataAccess importMissionsCodes should import Missions codes 1`] = `
Array [
  "BBTO",
  "BICH",
  "TAPI",
  "NANI",
  "NATO",
  "NEGE",
  "NELY",
  "NEMO",
  "NOTE",
  "OKLA",
  "OPEN",
  "QHAR",
  "QIKI",
  "QIWI",
  "QUDO",
  "QURE",
  "QURI",
  "QYAN",
  "QYLT",
  "QYRO",
  "TAXE",
  "TEDI",
  "TERI",
  "TETE",
  "TROC",
  "UPAC",
  "UPUS",
  "UZAR",
  "UZEL",
  "YRZZ",
  "ZEBU",
  "ZEMA",
  "ZEUS",
  "ZULI",
  "ZUPE",
  "ZITA",
  "ZBRA",
  "ZCAR",
  "ONIX",
  "QCOQ",
  "QNOR",
  "QBIK",
]
`;

exports[`dataAccess importMissionsSchedules should import Missions Schedules 1`] = `
Array [
  Object {
    "_metadata": Object {
      "call": "GET /missions/rers/A/NANI",
      "date": "2020-04-03T18:41:38+02:00",
      "version": 4,
    },
    "mission": "NANI",
    "result": Object {
      "stations": Array [
        Object {
          "name": "Cergy-Le Haut",
          "slug": "cergy+le+haut",
        },
        Object {
          "name": "Cergy-St-Christophe",
          "slug": "cergy+st+christophe",
        },
        Object {
          "name": "Cergy-Prefecture",
          "slug": "cergy+prefecture",
        },
        Object {
          "name": "Neuville Universite",
          "slug": "neuville+universite",
        },
        Object {
          "name": "Conflans-Fin d'Oise",
          "slug": "conflans+fin+d'oise",
        },
        Object {
          "name": "Acheres Ville",
          "slug": "acheres+ville",
        },
        Object {
          "name": "Maisons-Laffitte",
          "slug": "maisons+laffitte",
        },
        Object {
          "name": "Sartrouville",
          "slug": "sartrouville",
        },
        Object {
          "name": "Houilles",
          "slug": "houilles",
        },
        Object {
          "name": "Nanterre Prefecture",
          "slug": "nanterre+prefecture",
        },
        Object {
          "name": "Grande Arche la Defense",
          "slug": "grande+arche+la+defense",
        },
        Object {
          "name": "Charles de Gaulle-Etoile",
          "slug": "charles+de+gaulle+etoile",
        },
        Object {
          "name": "Auber",
          "slug": "auber",
        },
        Object {
          "name": "Chatelet-Les-Halles",
          "slug": "chatelet+les+halles",
        },
        Object {
          "name": "Gare de Lyon",
          "slug": "gare+de+lyon",
        },
        Object {
          "name": "Nation",
          "slug": "nation",
        },
        Object {
          "name": "Vincennes",
          "slug": "vincennes",
        },
        Object {
          "name": "Fontenay Sous Bois",
          "slug": "fontenay+sous+bois",
        },
        Object {
          "name": "Nogent Sur Marne",
          "slug": "nogent+sur+marne",
        },
        Object {
          "name": "Joinville le Pont",
          "slug": "joinville+le+pont",
        },
        Object {
          "name": "Saint Maur-Creteil",
          "slug": "saint+maur+creteil",
        },
        Object {
          "name": "Le Parc de Saint Maur",
          "slug": "le+parc+de+saint+maur",
        },
        Object {
          "name": "Champigny",
          "slug": "champigny",
        },
        Object {
          "name": "La Varenne-Chennevieres",
          "slug": "la+varenne+chennevieres",
        },
        Object {
          "name": "Sucy-Bonneuil",
          "slug": "sucy+bonneuil",
        },
        Object {
          "name": "Boissy Saint Leger",
          "slug": "boissy+saint+leger",
        },
      ],
    },
  },
  Object {
    "_metadata": Object {
      "call": "GET /missions/rers/A/NOTE",
      "date": "2020-04-03T18:42:30+02:00",
      "version": 4,
    },
    "mission": "NOTE",
    "result": Object {
      "stations": Array [
        Object {
          "name": "Poissy",
          "slug": "poissy",
        },
        Object {
          "name": "Acheres-Grand-Cormier",
          "slug": "acheres+grand+cormier",
        },
        Object {
          "name": "Maisons-Laffitte",
          "slug": "maisons+laffitte",
        },
        Object {
          "name": "Sartrouville",
          "slug": "sartrouville",
        },
        Object {
          "name": "Houilles",
          "slug": "houilles",
        },
        Object {
          "name": "Nanterre Prefecture",
          "slug": "nanterre+prefecture",
        },
        Object {
          "name": "Grande Arche la Defense",
          "slug": "grande+arche+la+defense",
        },
        Object {
          "name": "Charles de Gaulle-Etoile",
          "slug": "charles+de+gaulle+etoile",
        },
        Object {
          "name": "Auber",
          "slug": "auber",
        },
        Object {
          "name": "Chatelet-Les-Halles",
          "slug": "chatelet+les+halles",
        },
        Object {
          "name": "Gare de Lyon",
          "slug": "gare+de+lyon",
        },
        Object {
          "name": "Nation",
          "slug": "nation",
        },
        Object {
          "name": "Vincennes",
          "slug": "vincennes",
        },
        Object {
          "name": "Fontenay Sous Bois",
          "slug": "fontenay+sous+bois",
        },
        Object {
          "name": "Nogent Sur Marne",
          "slug": "nogent+sur+marne",
        },
        Object {
          "name": "Joinville le Pont",
          "slug": "joinville+le+pont",
        },
        Object {
          "name": "Saint Maur-Creteil",
          "slug": "saint+maur+creteil",
        },
        Object {
          "name": "Le Parc de Saint Maur",
          "slug": "le+parc+de+saint+maur",
        },
        Object {
          "name": "Champigny",
          "slug": "champigny",
        },
        Object {
          "name": "La Varenne-Chennevieres",
          "slug": "la+varenne+chennevieres",
        },
        Object {
          "name": "Sucy-Bonneuil",
          "slug": "sucy+bonneuil",
        },
        Object {
          "name": "Boissy Saint Leger",
          "slug": "boissy+saint+leger",
        },
      ],
    },
  },
]
`;
